### ISSUE TITLE
Tablet UI Extras

### ISSUE THREAD

**Issue by severtheskyline** [OxygenCobalt/Auxio#623]
### Description

Now that Android itself (And a lot of tablet apps) are adapted to a Tablet UI, I feel that Auxio could also implement some extra adaptations for tablets (Also I guess foldables, but those usually go hand in hand, so I'll just refer to both as tablets). Right now, Auxio's UI is pretty well adapted to tablets and I wouldn't say that, if these don't get implemented UX would be dead, but there are something things that would nicely fit on tablets. 

### Problem solved

With that being said, here are my main ideas that could improve the UI for tablet.
First, we have the main screen, which on a tablet looks something like this:
![Screenshot_20231123-111846_Trebuchet](https://github.com/OxygenCobalt/Auxio/assets/79670265/12cbf92d-99a3-4eee-976e-537ab8f7d38a)
It's not bad, however, 2 things really jump to me, the first is the shuffle button that's too big, but that's most likely a bug so that probably needs it's own issue. The main one is the top buttons, they looks too big and separated, but also (this may be nitpicking) the top bar is off-center, this works on a phone, but I feel that, in a tablet, it would be better centered and a bit mapper and compact (pictures of how other apps usually handle this in the next section).

Next, it hast to be the settings, which looks something like this:
![Screenshot_20231123-112238_Trebuchet](https://github.com/OxygenCobalt/Auxio/assets/79670265/3c4d4f07-8a1a-40a2-b497-b064fb7592ab)
Which again, looks fine, however (and this probably is pretty hard to implement, and if not implented it would be fine) is that the settings be a dual-pane layout, the pop-ups (like the one for excluding folders) would still remain pop-ups, but the settings that have 2 screens would be really nice as a dual-pane UI.

Lastly, we got the Playlist, which looks like this:
![Screenshot_20231123-112458_Auxio](https://github.com/OxygenCobalt/Auxio/assets/79670265/d835a982-d3c1-48d7-962b-7385153b5a99)
Which I'd say it is already adapted, but, continuing on dual-pane layouts, I feel that this UI could also benefit from a dual pane layout (screenshots on how another music player handles this).

### Other implementations

I got 2 examples which (doesn't have to and hopefully it doesn't look like them) can be used to get an idea on how the top bar may look. We first have newpipe:
![Screenshot_20231123-111954_Trebuchet](https://github.com/OxygenCobalt/Auxio/assets/79670265/defb4509-0b4d-4fec-8711-e3ce7b16fcb5)
Which is centered and compact, however, it only has icons with no labels and, realistically, they are unreachable on landscape view when holding up the tablet, but can give an idea on how it may look. The second one is Infinity for Reddit:
![Screenshot_20231123-112121_Trebuchet](https://github.com/OxygenCobalt/Auxio/assets/79670265/e4010b90-a206-4d6e-98de-b4172bed2722)
In this case, it is only labels, but it is a bit more separated, and I feel that, with icons, the UI could be ideal.

Next, with the settings, we don't even have to go that far in the OS, we can go to the system settings:
![Screenshot_20231123-120442_Configuración](https://github.com/OxygenCobalt/Auxio/assets/79670265/251818ff-6b34-4925-8375-0dc9356c3cab)
A dual-pane layout that let's you quickly change anything in settings and it not looking stretched out. The next example could be Tachiyomi:
![Screenshot_20231123-120638_Trebuchet](https://github.com/OxygenCobalt/Auxio/assets/79670265/81132c11-75fd-447b-8364-472818c5f575)
Which, I'd say it looks a bit weird, but the idea is the dual-pane settings layout. Also, if we want to stick with music players, we got YouTube Music:
![Screenshot_20231123-120848_Trebuchet](https://github.com/OxygenCobalt/Auxio/assets/79670265/51f9e2dd-a68c-46a8-ae27-5ac5413b3714)

And with the Playlist, again, I feel that they are fine if they are really hard to implement (or if you disagree that a dual-pane layout doesn't work here), the best example here is YouTube Music:
![Screenshot_20231123-112543_YouTube Music](https://github.com/OxygenCobalt/Auxio/assets/79670265/3bf1b9ef-c001-492b-98a3-8360d4105b15)
You get the Playlist controls and name at the left, with the content to the right. 

### Benefit

Having these Tablet UI Adaptations (or extras I guess) would benefit with a better UX when using a tablet/foldable. It also would mean a better/more consistent UI when comparing to other apps (even with Android itself, albeit the latest versions). 

### Duplicates

- [X] I have checked this feature request for duplicates.
- [X] I have checked that this feature is not implemented in the [lastest version](https://github.com/OxygenCobalt/Auxio/releases).
- [X] I have checked the [Why Are These Features Missing?](https://github.com/OxygenCobalt/Auxio/wiki/Why-Are-These-Features-Missing%3F) page.
- [X] I agree to the [Contribution Guidelines](https://github.com/OxygenCobalt/Auxio/blob/dev/.github/CONTRIBUTING.md).

**Comment by OxygenCobalt** [comment]
> It's not bad, however, 2 things really jump to me, the first is the shuffle button that's too big, but that's most likely a bug so that probably needs it's own issue. The main one is the top buttons, they looks too big and separated, but also (this may be nitpicking) the top bar is off-center, this works on a phone, but I feel that, in a tablet, it would be better centered and a bit mapper and compact (pictures of how other apps usually handle this in the next section).

The shuffle button is actually intended to be large. It's convention to use a large floating action button tablets, mostly so it's easier to press. As for the tabs, I wanted to make sure they were also easy to press, albeit I would agree that they would be better centered.

> Which again, looks fine, however (and this probably is pretty hard to implement, and if not implented it would be fine) is that the settings be a dual-pane layout, the pop-ups (like the one for excluding folders) would still remain pop-ups, but the settings that have 2 screens would be really nice as a dual-pane UI.

There's probably an off-the-shelf way I can do this? But in all likelihood it's riddled with bugs. I'll see what Tachiyomi does, theirs looks nearly identical to the system settings app.

> Which I'd say it is already adapted, but, continuing on dual-pane layouts, I feel that this UI could also benefit from a dual pane layout (screenshots on how another music player handles this).

I was actually considering making the detail view a dual-panel layout. This is actually pretty easy for me.

**Comment by severtheskyline** [comment]
> There's probably an off-the-shelf way I can do this? But in all likelihood it's riddled with bugs. I'll see what Tachiyomi does, theirs looks nearly identical to the system settings app.
I've seen Android documentation of it floating around, for compose at least, there is official ways to do it tho.

> I was actually considering making the detail view a dual-panel layout. This is actually pretty easy for me.
I guess I +1 that one lol

**Comment by severtheskyline** [comment]
_Ignore that notification you most likely got, I don't know how to markdown apparently_

Regarding the dual-pane UI, I've seen some official Android documentation for it, for compose at least, there are official ways to do it tho.

The rest, yeah, it's fine, I +1 the Playlists change as well lol

**Comment by OxygenCobalt** [comment]
> Regarding the dual-pane UI, I've seen some official Android documentation for it, for compose at least, there are official ways to do it tho.

I don't use compose, but I think there's at least a SlidingPanelView that does like...40% of this. In practice it will take a lot of wrangling to get working properly.

**Comment by jonxduran** [comment]
Also chiming in as a Galsxy Fold user, it would be really great if you could set the 2 column view in Now Playing to appear after a certain screen width, instead of only when device is in landscape. 

The reason is that Galaxy Folds unfold to portrait view, and still see the pull up drawer for queue even though the unfolded portrait Now Playing screen definitely has enough vertical space for the landscape 2 column view.

**Comment by OxygenCobalt** [comment]
> Also chiming in as a Galaxy Fold user, it would be really great if you could set the 2 column view in Now Playing to appear after a certain screen width, instead of only when device is in landscape.
> 
> The reason is that Galaxy Folds unfold to portrait view, and still see the pull up drawer for queue even though the unfolded portrait Now Playing screen definitely has enough vertical space for the landscape 2 column view.

Can you send a screenshot? It's is indeed based on largest screen width. I've yet to add real foldable support since it's a bit difficult (I would need to have a sliding dual-panel playback panel on some form factors, and a dual-panel playback/library view on others) @jonxduran

**Comment by jonxduran** [comment]
Sure here are 2 screenshots. Maybe switch from using the max of the 2 orientations to a defined minimum width, so both orientations of a foldable or tablet gets the 2 column Now Playing?

![portrait](https://github.com/OxygenCobalt/Auxio/assets/15164908/528be2b4-a5dc-4326-ab87-c77ce8a96992)

![landscape](https://github.com/OxygenCobalt/Auxio/assets/15164908/cef87942-8c00-467e-8271-39aba4f4b07d)

**Comment by OxygenCobalt** [comment]
Ah, I see now @jonxduran. I think I can just move the layout from `w600dp-land` to `w600dp`. Thanks for pointing this out.

**Comment by OxygenCobalt** [comment]
Quick question: What do you all think about the current playback bar (the thing that sits on the bottom of the screen when you swipe down the playback view) @severtheskyline @jonxduran?  As far as I'm aware, it's just the phone layout stretched across the entire span of the tablet screen, which wastes a ton of space and is probably a but clunky to use. Do you have any ideas on what would have it better?

**Comment by severtheskyline** [comment]
> Quick question: What do you all think about the current playback bar (the thing that sits on the bottom of the screen when you swipe down the playback view) @severtheskyline @jonxduran? It's just the phone layout stretched across the entire span of the tablet screen, which wastes a ton of space and is probably a bit clunky to use. Do you have any ideas on what would make it better, if any?

I guess you mean this one 

![Screenshot_20231211-131126_Auxio](https://github.com/OxygenCobalt/Auxio/assets/79670265/deb24c6b-70e3-48d0-8b79-0b54ef18436f)

Honestly, I feel that it could be better, but I have no idea specifically on how it would be, like, every other music player has the same player, except Spotify but that doesn't look good. Maybe a mini-player like YouTube on a tablet? Just an idea, since I don't really know how well would it look, if it's even possible to do at all

![Screenshot_20231211-142957_YouTube](https://github.com/OxygenCobalt/Auxio/assets/79670265/d47e3630-fdc4-447d-8abd-3ecf2ca42de8)

I'd guess it would be squared because of how album art is. Another idea would be to show the next song in queue alongside what's playing.

The other ideas could apply to both phones and and tablets, like being able to swipe to the right or left the mini player, but I guess that would be it's own thing.

**Comment by OxygenCobalt** [comment]
My idea currently is to just center it (like you want with the tabs) and add some additional controls. @severtheskyline

**Comment by OxygenCobalt** [comment]
Yeah, just taking the existing layout but removing that huge gap and centering it. @severtheskyline 

Also: I've come around on reducing the size of some of the UI components (the shuffle button and tabs). Now that I really think about it, it's just too inconsistent. I'll make them smaller.

**Comment by severtheskyline** [comment]
Do you have any screenshots of those UI components that got changed? Just to get idea on how they look at the moment

**Comment by OxygenCobalt** [comment]
I actually haven't changed them, I was just trying to get ideas. I'll send some when I can get time.

**Comment by OxygenCobalt** [comment]
@severtheskyline Update: Got a new detail panel going, thoughts?

![image](https://github.com/OxygenCobalt/Auxio/assets/65370175/6d10ddbe-f9ce-4880-80b8-66122135d85c)

**Comment by severtheskyline** [comment]
> @severtheskyline Update: Got a new detail panel going, thoughts?
> 
> ![image](https://private-user-images.githubusercontent.com/65370175/346252414-6d10ddbe-f9ce-4880-80b8-66122135d85c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjExNjY0NjEsIm5iZiI6MTcyMTE2NjE2MSwicGF0aCI6Ii82NTM3MDE3NS8zNDYyNTI0MTQtNmQxMGRkYmUtZjljZS00ODgwLTgwYjgtNjYxMjIxMzVkODVjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA3MTYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNzE2VDIxNDI0MVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWY1ZTQ4ZThjYjMwNTZiNzdhMjQyOTkyZjg0NzlkZjI0YjU0NDA4M2U5Njg0MDg3ZjUzOWU0OWE4MDEwODhlZGUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.4egcnW9fe2unAmEsEnn2YbpCu2IKTibCHbtaEDrSgT0)

That looks awesome, I would probably add something that makes more clear the division between the album/playlist details and the songs themselves, but if that was the final UI, I think it looks great. Do you have an screenshot off how it looks in dark mode?

_sorry for the late response_

**Comment by jonxduran** [comment]
> Ah, I see now [@jonxduran](https://github.com/jonxduran). I think I can just move the layout from `w600dp-land` to `w600dp`. Thanks for pointing this out.

Just popping in here again to say I love this change for my Fold 5! This is how it looks for me now in portrait unfolded mode:

![Image](https://github.com/user-attachments/assets/ef4a92ac-f754-44c3-9eea-60be60bc90e6)

**Comment by TL-P** [comment]
> [@severtheskyline](https://github.com/severtheskyline) Update: Got a new detail panel going, thoughts?
> 
> ![image](https://github.com/OxygenCobalt/Auxio/assets/65370175/6d10ddbe-f9ce-4880-80b8-66122135d85c)

I have been using Auxio on a tablet recently and just wanted to say that this looks great, much better than what there is currently in my opinion!

**Comment by OxygenCobalt** [comment]
I agree @TL-P, also better for foldables. I'll see if I can find (or just re-create) this draft

**Comment by TL-P** [comment]
@OxygenCobalt I found a weird bug that might belong here: On my tablet, these two buttons on the bottom right on the playlist page don't seem to do anything.

![Image](https://github.com/user-attachments/assets/de0d92da-ec66-43a7-b746-9dbd88b77cee)

It is probably just an UI issue since you can still create a playlist with the button in the middle and then import in the dropdown menu:

![Image](https://github.com/user-attachments/assets/f8a9280e-a3dd-486b-81aa-45fcff49f3b1)

The buttons work fine on my phone so I assume this is a tablet UI issue?

**Comment by OxygenCobalt** [comment]
Yeah guess so, probably some click area conflict or mismatch of some kind, will fix @TL-P

**Comment by TL-P** [comment]
Edit: also does not work on my phone when in landscape mode

**Comment by TL-P** [comment]
> [@OxygenCobalt](https://github.com/OxygenCobalt) I found a weird bug that might belong here: On my tablet, these two buttons on the bottom right on the playlist page don't seem to do anything.
> 
> ![Image](https://github.com/user-attachments/assets/de0d92da-ec66-43a7-b746-9dbd88b77cee)
> 
> It is probably just an UI issue since you can still create a playlist with the button in the middle and then import in the dropdown menu:
> 
> ![Image](https://github.com/user-attachments/assets/f8a9280e-a3dd-486b-81aa-45fcff49f3b1)
> 
> The buttons work fine on my phone so I assume this is a tablet UI issue?

Decided to test v3.6.3 on the tablet and the buttons seem to work there.

**Comment by OxygenCobalt** [comment]
Let me look into this @TL-P

**Comment by TL-P** [comment]
Since you are working on the UI for v4.1, maybe this issue could be done alongside the other changes? :)

**Comment by OxygenCobalt** [comment]
Sorry for missing this @TL-P. I can consider, one of the unsolved issues was persisting scroll/collapse states between rotation. 

At the same time I also want to deliver v4.1 sooner (I'm aiming for today) but I can add it going forward.

**Comment by TL-P** [comment]
No worries @OxygenCobalt, I'm grateful you're doing this in your spare time :)

Probably better to release v4.1 first, and then do this. I am happy to test anything to see if it works on my device.

**Comment by OxygenCobalt** [comment]
Yeah, decided to add it @TL-P. Done now.

**Comment by OxygenCobalt** [comment]
This is completed now
